aperturedb/postgres.go

```diff
@@ -0,0 +1,140 @@
+package aperturedb
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+	"time"
+
+	postgres_migrate "github.com/golang-migrate/migrate/v4/database/postgres"
+	// Import the file source to register it with the migrate library.
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	"github.com/lightninglabs/aperture/aperturedb/sqlc"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	dsnTemplate = "postgres://%v:%v@%v:%d/%v?sslmode=%v"
+)
+
+var (
+	// DefaultPostgresFixtureLifetime is the default maximum time a Postgres
+	// test fixture is being kept alive. After that time the docker
+	// container will be terminated forcefully, even if the tests aren't
+	// fully executed yet. So this time needs to be chosen correctly to be
+	// longer than the longest expected individual test run time.
+	DefaultPostgresFixtureLifetime = 10 * time.Minute
+)
+
+// PostgresConfig holds the postgres database configuration.
+type PostgresConfig struct {
+	SkipMigrations     bool   `long:"skipmigrations" description:"Skip applying migrations on startup."`
+	Host               string `long:"host" description:"Database server hostname."`
+	Port               int    `long:"port" description:"Database server port."`
+	User               string `long:"user" description:"Database user."`
+	Password           string `long:"password" description:"Database user's password."`
+	DBName             string `long:"dbname" description:"Database name to use."`
+	MaxOpenConnections int32  `long:"maxconnections" description:"Max open connections to keep alive to the database server."`
+	RequireSSL         bool   `long:"requiressl" description:"Whether to require using SSL (mode: require) when connecting to the server."`
+}
+
+// DSN returns the dns to connect to the database.
+func (s *PostgresConfig) DSN(hidePassword bool) string {
+	var sslMode = "disable"
+	if s.RequireSSL {
+		sslMode = "require"
+	}
+
+	password := s.Password
+	if hidePassword {
+		// Placeholder used for logging the DSN safely.
+		password = "****"
+	}
+
+	return fmt.Sprintf(dsnTemplate, s.User, password, s.Host, s.Port,
+		s.DBName, sslMode)
+}
+
+// PostgresStore is a database store implementation that uses a Postgres
+// backend.
+type PostgresStore struct {
+	cfg *PostgresConfig
+
+	*BaseDB
+}
+
+// NewPostgresStore creates a new store that is backed by a Postgres database
+// backend.
+func NewPostgresStore(cfg *PostgresConfig) (*PostgresStore, error) {
+	log.Infof("Using SQL database '%s'", cfg.DSN(true))
+
+	rawDB, err := sql.Open("pgx", cfg.DSN(false))
+	if err != nil {
+		return nil, err
+	}
+
+	maxConns := defaultMaxConns
+	if cfg.MaxOpenConnections > 0 {
+		maxConns = int(cfg.MaxOpenConnections)
+	}
+
+	rawDB.SetMaxOpenConns(maxConns)
+	rawDB.SetMaxIdleConns(maxConns)
```

Comment:
We need to actually set the number of maximum connections. Also, maybe it makes sense to take the other fixes applied here https://github.com/lightninglabs/taproot-assets/pull/336 and here https://github.com/lightninglabs/taproot-assets/pull/334

Reply:
`We need to actually set the number of maximum connections.` i do not know what this means. Aren't we doing that with `rawDB.SetMaxOpenConns(maxConns)`?

I also added the changes from tapdb, thanks for pointing to them :+1:

Reply:
Sorry, my wording was a bit confusing. I meant that we have a config flag for Postgres that allows the user to specify the number of connections. But here we just set the constant, so the config flag is unused.